rfcs/0000-changes-lifecyle.md

```diff
@@ -0,0 +1,205 @@
+
+# Motivation
+
+[motivation]: #motivation
+
+The compiler needs a guide on how to introduce changes to improve developer experience every time a compiler is updated.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+## PR changes
+
+Every PR will have a section named "Release Notes", which will contains a user friendly explanation of what the PR contains.
+
+The compiler release notes **WILL** be the amalgamation of these section "as is".
+
+## Feature flags
+
+Any complex change that needed to be gated will need the following steps.
+
+1 - A specific github issue labeled with `feature-*` and `track-feature`. This issue should be the umbrella for everything related with this feature;
+2 - A preliminary PR enabling the feature flag should be created and linked in to the umbrella issue; This PR will enable the feature flag `--experimental <comma-separated-list>` on all tools.
+3 - As many PRs will be created and merged an normal;
+4 - When the feature is ready, a closing PR will be created and wait until the feature flag is enabled by default.
+5 - On a later date, the feature flag can be removed making the feature the default behavior of the compiler. 
+
+# Enabling features on `sway`
+
+Features can be enabled inside the `Forc.toml` file:
+
+```toml
+experimental = ["...", "..."]
+```
+
+or using the CLI
+
+```
+> forc ... --experimental some_feature,another_feature
+```
+
+These flags also need to be enabled programmatically by any compiler driver, like tests.
+
+Unlike `Rust` we will not support features inside sway code like the example below, because some features will span across multiple tools. That would demand `forc` to parse, or ask the `sway` compiler if a feature is enabled or not. 
+
+```sway
+#![enable(some_experimental_feature)]
+```
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+This RFC intends to propose a scheme to track changes in the compiler. These changes will be categorized into two ways:
+
+1 - Non-breaking changes;
+2 - Breaking changes.
+
+Breaking changes will be any changes that, after updating the compiler version, causes:
+
+1 - Code without deprecated warning to be flagged as error;
+2 - ABI json properties to be removed or have their type changed;
+3 - SDK (rust, typescript and others) to break;
+4 - Receipt parsers to break;
+
+The following changes are **not** considered to be breaking changes:
+
+1 - Code flagged with deprecated warning to be flagged as error;
+2 - Code that required an experimental flag, but now are the default behavior to compile;
+3 - Contract ID to change;
+4 - new ABI json properties;
+
+## Minor Version Updates
+
+When a developer updates to a different minor version, it is expected no breaking changes, except for severe bugs.
+
+Minor changes will consider these to be breaking and non-breaking changes:
+
+### Breaking Changes 
+
```

Comment:
It might be good to start off this section with a definition of breaking changes, before listing the categories of breaking changes. Agreeing on a definition will make it easier to debate the categorization.

Here's a definition to start with.

```suggestion
### Breaking Changes 

A breaking change is a difference in functionality from the previous version of the compiler that may require an update to sway code in order for it to compile.
```